hera_qm/omnical_metrics.py

```diff
@@ -12,3 +28,796 @@ def get_omnical_metrics_dict():
+            for each antenna's chisq variability.
+        """
+        # assign fc filename
+        self.firstcal_file = firstcal_file
+
+        # select freq channels
+        if cut_band == True:
+            self.band = np.arange(self.Nfreqs/10, self.Nfreqs - self.Nfreqs/10)
+        else:
+            self.band = np.arange(self.Nfreqs)
+
+        # Get robust standard deviation of chisq for each antenna
+        chisq_ant_std       = np.sqrt(np.array(map(astats.biweight_midvariance, self.chisq[:, :, self.band, :])))
+        chisq_ant_std_loc   = astats.biweight_location(chisq_ant_std)
+        chisq_ant_std_scale = np.sqrt(astats.biweight_midvariance(chisq_ant_std))
+
+        # Pick out "bad" antennas from chisq_std_cut, which is a cut in the standard deviation
+        # of the chisq fluctuations across time, frequency and polarizations
+        chisq_bad_ants = self.ant_array[np.where((chisq_ant_std-chisq_ant_std_loc) > chisq_ant_std_scale * chisq_std_cut)]
+
+        # convert to dictionaries
+        chisq_ant_std = OrderedDict(zip(self.ant_array, chisq_ant_std))
+
+        if firstcal_file is not None and hasattr(self, 'firstcal_gains') == False:
+            # load fc gain solutions
+            self.firstcal_delays, self.firstcal_gains = load_firstcal_gains(firstcal_file)
+
+            # get gain difference between omnical and firstcal gain solutions
+            self.gain_diff = self.omni_gains * self.firstcal_gains.conj()
+
+            # run phs FT metric
+            (ant_phs_noise, tot_phs_noise, phs_noise_bad_ants,
+            phs_noise_good_sol) = self.phs_FT_metric(np.angle(self.gain_diff), phs_noise_cut=phs_noise_cut)
+
+            # run phs std metric
+            (ant_phs_std, tot_phs_std, phs_std_bad_ants, 
+            phs_std_good_sol) = self.phs_std_metric(np.angle(self.gain_diff), phs_std_cut=phs_std_cut)
+
+        # initialize metrics
+        metrics                        = OrderedDict()
+
+        metrics['chisq_ant_std']       = chisq_ant_std
+        metrics['chisq_ant_std_loc']   = chisq_ant_std_loc
+        metrics['chisq_ant_std_scale'] = chisq_ant_std_scale
+        metrics['chisq_bad_ants']      = chisq_bad_ants
+        metrics['chisq_std_cut']       = chisq_std_cut
+
+        metrics['freqs']               = self.freqs
+        metrics['Nfreqs']              = self.Nfreqs
+        metrics['cut_band']            = cut_band
+        metrics['band']                = self.band
+        metrics['ant_array']           = self.ant_array
+        metrics['jones']               = self.jones
+        metrics['pols']                = self.pols
+        metrics['Npols']               = self.Npols
+        metrics['times']               = self.times
+        metrics['Ntimes']              = self.Ntimes
+        metrics['Nants']               = self.Nants
+
+        metrics['version']             = self.version_str
+        metrics['history']             = self.history
```

Comment:
As far as I can tell, this is always an empty string.

Reply:
Oh, I saw how you update it in omnical_metrics_run. Could we maybe make an optional input the the init function? It's a bit odd to have the user populate an attribute like this manually after the fact.